quickwit-metastore/src/tests.rs

```diff
@@ -2170,6 +2172,397 @@ pub mod test_suite {
+            split_id: split_id_2.to_string(),
+            num_docs: 1,
+            uncompressed_docs_size_in_bytes: 2,
+            create_timestamp: current_timestamp,
+            delete_opstamp: 10,
+            ..Default::default()
+        };
+        let split_id_3 = "list-stale-splits-three";
+        let split_metadata_3 = SplitMetadata {
+            footer_offsets: 1000..2000,
+            split_id: split_id_3.to_string(),
+            num_docs: 1,
+            uncompressed_docs_size_in_bytes: 2,
+            create_timestamp: current_timestamp,
+            delete_opstamp: 0,
+            ..Default::default()
+        };
+
+        {
+            info!("List stale splits on a non-existent index");
+            let metastore_err = metastore
+                .list_stale_splits("non-existent-index", 0, 10)
+                .await
+                .unwrap_err();
+            error!(err=?metastore_err);
+            assert!(matches!(
+                metastore_err,
+                MetastoreError::IndexDoesNotExist { .. }
+            ));
+        }
+
+        {
+            info!("List stale splits on an index");
+            metastore
+                .create_index(index_metadata.clone())
+                .await
+                .unwrap();
+
+            metastore
+                .stage_split(index_id, split_metadata_1.clone())
+                .await
+                .unwrap();
+
+            metastore
+                .stage_split(index_id, split_metadata_2.clone())
+                .await
+                .unwrap();
+
+            metastore
+                .stage_split(index_id, split_metadata_3.clone())
+                .await
+                .unwrap();
+            metastore
+                .publish_splits(index_id, &[split_id_1, split_id_2], &[], None)
+                .await
+                .unwrap();
+
+            let splits = metastore.list_stale_splits(index_id, 100, 1).await.unwrap();
+            assert_eq!(splits.len(), 1);
+            assert_eq!(
+                splits[0].split_metadata.delete_opstamp,
+                split_metadata_2.delete_opstamp
+            );
+
+            let splits = metastore.list_stale_splits(index_id, 100, 2).await.unwrap();
+            assert_eq!(splits.len(), 2);
+            assert_eq!(
+                splits[1].split_metadata.delete_opstamp,
+                split_metadata_1.delete_opstamp
+            );
+
+            let splits = metastore.list_stale_splits(index_id, 20, 2).await.unwrap();
+            assert_eq!(splits.len(), 1);
+            assert_eq!(
+                splits[0].split_metadata.delete_opstamp,
+                split_metadata_2.delete_opstamp
+            );
+
+            let splits = metastore.list_stale_splits(index_id, 10, 2).await.unwrap();
+            assert!(splits.is_empty());
+            cleanup_index(&metastore, index_id).await;
+        }
+    }
+
+    pub async fn test_metastore_update_splits_delete_opstamp<
```

Comment:
please add a unit test to see what happens when  we delete an index with delete tasks.

Reply:
Done: `test_metastore_delete_index_with_tasks`.